File: app/prototypes/1/android/RumpelAndroid/app/src/main/java/com/rumpel/rumpelandroid/activities/LoginActivity.java
package com.rumpel.rumpelandroid.activities;

import android.content.Intent;
import android.os.Bundle;
import android.text.InputType;
import android.text.SpannableString;
import android.text.style.UnderlineSpan;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.rumpel.rumpelandroid.R;
import com.rumpel.rumpelandroid.realm.DAOUser;
import com.rumpel.rumpelandroid.utils.AndroidUtils;

public class LoginActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        TextView signUpLogin = findViewById(R.id.switchToSignUpBtn);
        ProgressBar loading = findViewById(R.id.loading);
        SpannableString spString = new SpannableString(signUpLogin.getText());
        spString.setSpan(new UnderlineSpan(), 0, spString.length(), 0);
        signUpLogin.setText(spString);
        signUpLogin.setOnClickListener(v -> startActivity(new Intent(LoginActivity.this, SignUpActivity.class)));
        DAOUser dao = new DAOUser(this);
        EditText usernameTxt = findViewById(R.id.username);
        EditText passTxt = findViewById(R.id.password);

        usernameTxt.setText("username");

        Button loginBtn = findViewById(R.id.loginBtn);
        loginBtn.setOnClickListener(v -> {
            boolean authenticate = dao.authenticate(usernameTxt.getText().toString(), passTxt.getText().toString());
            if (authenticate) {
                Log.v("MONGO", "Authenticated!");
                startActivity(new Intent(LoginActivity.this, HomeActivity.class));
                finish();
            } else AndroidUtils.toast(LoginActivity.this, getString(R.string.failed_to_authenticate));
            loading.setVisibility(View.INVISIBLE);
        });

        CheckBox checkBox = findViewById(R.id.checkBox);
        final int inputType = passTxt.getInputType();
        checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) passTxt.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS);
            else passTxt.setInputType(inputType);
        });


    }

}

File: app/prototypes/2/android/Rumpel/app/src/main/java/com/rumpel/rumpelandroid/ui/activities/LoginActivity.kt
package com.rumpel.rumpelandroid.ui.activities

import android.content.Intent
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.rumpel.rumpelandroid.db.DAOUser
import com.rumpel.rumpelandroid.ui.screens.LoginScreen

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        installSplashScreen()
        val dao = DAOUser(this)
        setContent {
            LoginScreen(onLogin = {username, pass -> dao.authenticate("username", "1martson1") // placeholder to save time
            }, switchToSignUp = {
                startActivity(Intent(this, SignUpActivity::class.java))
                finish()
            })
        }

    }
}